script.js

```diff
@@ -1,4 +1,200 @@
+
+// (game mode : choose dice order) The player picks the order of the dice they want. For example, if they wanted the number 63, they would specify that the 2nd dice goes first. You can choose how the player specifies dice order.
+// (second message) Player 1, you chose Dice 2 first. <br> Your number is 63. <br> It is now Player 2's turn.
+
+// (game mode : compare numbers and see who is winner) After both players have rolled and chosen dice order, the player with the higher combined number wins.
+
+// ***steps
+//1) player 1 rolls dice and return with the output of each dice roll result. + choose the dice order and place the combined number result
+//2) change to player 2 and play same as 1
+//3) compare player 1 result and player 2 result to compare and find out who is the winner
+//4) go back to beginning to reset the game without refreshing the HTML
+
+//global variables//
+var gameModeDiceRoll = "Game Mode : Dice Roll";
+var gameModeChooseDice = "Game Mode : Choose Dice";
+var gameModeCompareResult = "Game Mode : Compare Dice Results";
+var gameCurrentMode = gameModeDiceRoll;
+
+var currentPlayerRoll = [];
+
+var currentPlayer = 1;
+var allPlayerScores = [];
+
+//All functions here //
+var rollDice = function () {
+  console.log("check : random dice roll start!");
+  // produce a decimal between 0 and 6
+  var randomDecimal = Math.random() * 6;
+  // remove the decimal
+  var randomInteger = Math.floor(randomDecimal);
+  // add 1 to get a number between 1 and 6 inclusive
+  var diceNumber = randomInteger + 1;
+  console.log("Dice rolled result : " + diceNumber);
+  return diceNumber;
+};
+
+var rollDiceForPlayer = function () {
+  console.log("time to roll dice for player");
+  var counter = 0;
+  while (counter < 2) {
+    currentPlayerRoll.push(rollDice());
+    counter += 1;
+  }
+  console.log(
+    "played 2 times for player's dice roll data : " + currentPlayerRoll
+  );
+  return (
+    "Hello player" +
+    currentPlayer +
+    ", <br><br> " +
+    "You rolled <br>" +
+    "Dice 1 : " +
+    currentPlayerRoll[0] +
+    "<br>" +
+    "Dice 2 : " +
+    currentPlayerRoll[1] +
+    "<br><br>" +
+    "Now, please choose which dice you want to place as your placement. Input 1 for dice 1. Input 2 for dice 2. "
+  );
+};
+
+var getPlayerScore = function (input) {
+  var playerScore;
+
+  // input == 1
+  if (input == 1) {
+    console.log("input == 1st dice");
+    var playerScore = Number(
+      String(currentPlayerRoll[0]) + String(currentPlayerRoll[1])
+    );
+    console.log(playerScore);
+    // store this player score to all player score
+    allPlayerScores.push(playerScore);
+    //need to clean up current player roll array
+    currentPlayerRoll = [];
+    return (
+      "okay, you choose 1st dice to be placed first. <br>" +
+      "Your result : " +
+      playerScore
+    );
+  }
+
+  // input == 2
+  if (input == 2) {
```

Comment:
In scenarios where only 1 if statement should happen, you should use if and else if statements rather than 2 if statements. Reason being, if you use 2 if statements, it might seem that you might want more than 1 if statement to happen.